staging/src/k8s.io/pod-security-admission/policy/registry.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package policy
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+
+	"k8s.io/pod-security-admission/api"
+)
+
+// Registry holds the Checks that are used to validate a policy.
+type Registry interface {
+	// ChecksForIDAndVersion fetches the check with the given ID that should be evaluated for the version.
+	// An error is returned if no checks are registered with the given ID, or the given version is
+	// older than the first version of the check.
+	CheckForIDAndVersion(id string, version api.Version) (Check, error)
+	// ChecksForLevelAndVersion fetches all the checks that should be evaluated for the given level
+	// and version.
+	ChecksForLevelAndVersion(level api.Level, version api.Version) ([]Check, error)
+}
+
+// CheckRegistry provides a default implementation of a Registry.
+type CheckRegistry struct {
+	// The checks are a map of check_ID -> sorted slice of versioned checks, newest first
+	baselineChecks, restrictedChecks map[string][]versionedCheck
+}
+
+type versionedCheck struct {
+	firstVersion api.Version
+	Check
+}
+
+func NewCheckRegistry() *CheckRegistry {
+	return &CheckRegistry{
+		baselineChecks:   map[string][]versionedCheck{},
+		restrictedChecks: map[string][]versionedCheck{},
+	}
+}
+
+func (r *CheckRegistry) CheckForIDAndVersion(id string, version api.Version) (Check, error) {
```

Comment:
This method may give me a baseline check or a restricted check?  That surprises me a little bit, I guess I'll see how it's used.

Reply:
this is unused now, I think it was intended to allow access to a specific check for consumers you want to override/wrap

maybe remove until needed or replace with direct accessor to specific checks (GetSELinuxCheck(), etc)

Reply:
removed in reworked registry approach